From 0e65721c09e068bc7d795589eb4bf84d50e0497c Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Fri, 11 Aug 2017 14:06:06 -0400
Subject: [PATCH] Patched

---
 aeson-pretty.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/aeson-pretty.cabal b/aeson-pretty.cabal
index ac6d451..c62c6d1 100644
--- a/aeson-pretty.cabal
+++ b/aeson-pretty.cabal
@@ -33,7 +33,7 @@ extra-source-files:
 
 flag lib-only
     description: Only build/install the library, NOT the command-line tool.
-    default: False
+    default: True
 
 library
     exposed-modules:
-- 
2.7.4 (Apple Git-66)

