From d4db5cc80919953a3c2f5780867a4d88187295f7 Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Sat, 12 Aug 2017 21:05:28 -0400
Subject: [PATCH] Patched

---
 aeson-pretty.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/aeson-pretty.cabal b/aeson-pretty.cabal
index 2b69fc2..848aeb8 100644
--- a/aeson-pretty.cabal
+++ b/aeson-pretty.cabal
@@ -33,7 +33,7 @@ extra-source-files:
 
 flag lib-only
     description: Only build/install the library, NOT the command-line tool.
-    default: False
+    default: True
 
 library
     exposed-modules:
-- 
2.7.4 (Apple Git-66)

