From 99e914a0327c36e5b6511d6a7c3191bbd2ea5df1 Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Thu, 20 Jul 2017 22:40:11 -0400
Subject: [PATCH] Patched

---
 aeson-pretty.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/aeson-pretty.cabal b/aeson-pretty.cabal
index 30dd78b..69c62d9 100644
--- a/aeson-pretty.cabal
+++ b/aeson-pretty.cabal
@@ -33,7 +33,7 @@ extra-source-files:
 
 flag lib-only
     description: Only build/install the library, NOT the command-line tool.
-    default: False
+    default: True
 
 library
     exposed-modules:
-- 
2.7.4 (Apple Git-66)

