From b95868461d1fc04ea2caa85b5e32c9c662a1bb6a Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Mon, 2 Jan 2017 21:02:57 +0530
Subject: [PATCH] Patched

---
 ansi-terminal.cabal | 52 ++++++++++++++++++++++++++--------------------------
 1 file changed, 26 insertions(+), 26 deletions(-)

diff --git a/ansi-terminal.cabal b/ansi-terminal.cabal
index e6b56a6..b45fb47 100644
--- a/ansi-terminal.cabal
+++ b/ansi-terminal.cabal
@@ -34,20 +34,20 @@ Library
         Include-Dirs:           includes
         
         Build-Depends:          base >= 4 && < 5
-        if os(windows)
-                Build-Depends:          Win32 >= 2.0
-                Cpp-Options:            -DWINDOWS
-                Extra-Libraries:        "kernel32"
-                Other-Modules:          System.Console.ANSI.Windows
-                                        System.Console.ANSI.Windows.Foreign
-                                        System.Console.ANSI.Windows.Emulator
-                                        -- NB: used for fallback by the emulator
-                                        System.Console.ANSI.Unix
-        else
-                -- We assume any non-Windows platform is Unix
-                Build-Depends:          unix >= 2.3.0.0
-                Cpp-Options:            -DUNIX
-                Other-Modules:          System.Console.ANSI.Unix
+        -- if os(windows)
+        --         Build-Depends:          Win32 >= 2.0
+        --         Cpp-Options:            -DWINDOWS
+        --         Extra-Libraries:        "kernel32"
+        --         Other-Modules:          System.Console.ANSI.Windows
+        --                                 System.Console.ANSI.Windows.Foreign
+        --                                 System.Console.ANSI.Windows.Emulator
+        --                                 -- NB: used for fallback by the emulator
+        --                                 System.Console.ANSI.Unix
+        -- else
+        --         -- We assume any non-Windows platform is Unix
+        --         Build-Depends:          unix >= 2.3.0.0
+        Cpp-Options:            -DUNIX
+        Other-Modules:          System.Console.ANSI.Unix
         
         Extensions:             CPP
                                 ForeignFunctionInterface
@@ -59,18 +59,18 @@ Executable ansi-terminal-example
         
         Include-Dirs:           includes
         
-        if os(windows)
-                Build-Depends:          Win32 >= 2.0
-                Cpp-Options:            -DWINDOWS
-                Extra-Libraries:        "kernel32"
-                Other-Modules:          System.Console.ANSI.Windows
-                                        System.Console.ANSI.Windows.Foreign
-                                        System.Console.ANSI.Windows.Emulator
-        else
-                -- We assume any non-Windows platform is Unix
-                Build-Depends:          unix >= 2.3.0.0
-                Cpp-Options:            -DUNIX
-                Other-Modules:          System.Console.ANSI.Unix
+        -- if os(windows)
+        --         Build-Depends:          Win32 >= 2.0
+        --         Cpp-Options:            -DWINDOWS
+        --         Extra-Libraries:        "kernel32"
+        --         Other-Modules:          System.Console.ANSI.Windows
+        --                                 System.Console.ANSI.Windows.Foreign
+        --                                 System.Console.ANSI.Windows.Emulator
+        -- else
+        --         -- We assume any non-Windows platform is Unix
+        --         Build-Depends:          unix >= 2.3.0.0
+        Cpp-Options:            -DUNIX
+        Other-Modules:          System.Console.ANSI.Unix
         
         
         Build-Depends:          base >= 4 && < 5
-- 
2.7.4 (Apple Git-66)

