From 6edefdd806f6257756f6b2997a6e7048cfda3eeb Mon Sep 17 00:00:00 2001
From: jneira <atreyu.bbb@gmail.com>
Date: Sun, 6 Jan 2019 21:49:18 +0100
Subject: [PATCH] Patched

---
 dhall.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dhall.cabal b/dhall.cabal
index 1493db9..05f00f4 100644
--- a/dhall.cabal
+++ b/dhall.cabal
@@ -335,7 +335,7 @@ Library
         http-types                  >= 0.7.0    && < 0.13,
         http-client                 >= 0.4.30   && < 0.6 ,
         http-client-tls             >= 0.2.0    && < 0.4
-    if !impl(ghc >= 8.0)
+    if !impl(ghc >= 8.0) && !impl(eta >= 0.8.4)
       Build-Depends: semigroups == 0.18.*
       Build-Depends: transformers == 0.4.2.*
       Build-Depends: fail == 4.9.*
-- 
2.16.2.windows.1

