From b35a119dc2710e1be02e92d3b4e68dc7315f48d5 Mon Sep 17 00:00:00 2001
From: jneira <atreyu.bbb@gmail.com>
Date: Mon, 11 Jun 2018 12:23:56 +0200
Subject: [PATCH] Comment dhall-repl for now

---
 dhall.cabal | 32 ++++++++++++++++----------------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/dhall.cabal b/dhall.cabal
index 86000cb..258bb4c 100644
--- a/dhall.cabal
+++ b/dhall.cabal
@@ -213,22 +213,22 @@ Executable dhall
     Other-Modules:
         Paths_dhall
 
-Executable dhall-repl
-    Hs-Source-Dirs: dhall-repl
-    Main-Is: Main.hs
-    Build-Depends:
-        base             >= 4        && < 5   ,
-        ansi-terminal                         ,
-        dhall                                 ,
-        haskeline        >= 0.7.3.0  && < 0.8 ,
-        mtl              >= 2.2.1    && < 2.3 ,
-        repline          >= 0.1.6.0  && < 0.2 ,
-        prettyprinter                         ,
-        prettyprinter-ansi-terminal           ,
-        text
-    if !impl(ghc >= 8.0)
-      Build-Depends: transformers == 0.4.2.*
-    GHC-Options: -Wall
+-- Executable dhall-repl
+--     Hs-Source-Dirs: dhall-repl
+--     Main-Is: Main.hs
+--     Build-Depends:
+--         base             >= 4        && < 5   ,
+--         ansi-terminal                         ,
+--         dhall                                 ,
+--         haskeline        >= 0.7.3.0  && < 0.8 ,
+--         mtl              >= 2.2.1    && < 2.3 ,
+--         repline          >= 0.1.6.0  && < 0.2 ,
+--         prettyprinter                         ,
+--         prettyprinter-ansi-terminal           ,
+--         text
+--     if !impl(ghc >= 8.0)
+--       Build-Depends: transformers == 0.4.2.*
+--     GHC-Options: -Wall
 
 Executable dhall-format
     Hs-Source-Dirs: dhall-format
-- 
2.16.2.windows.1

