From 0c670b2f99fe6f446d060b8bcf9ff33d25091a72 Mon Sep 17 00:00:00 2001
From: jneira <atreyu.bbb@gmail.com>
Date: Wed, 17 Oct 2018 08:27:59 +0200
Subject: [PATCH] Patched

---
 dhall.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dhall.cabal b/dhall.cabal
index e8442d7..243b9d2 100644
--- a/dhall.cabal
+++ b/dhall.cabal
@@ -146,7 +146,7 @@ Library
       Build-Depends:
         http-client                 >= 0.4.30   && < 0.6 ,
         http-client-tls             >= 0.2.0    && < 0.4
-    if !impl(ghc >= 8.0)
+    if !impl(ghc >= 8.0) && !impl(eta >= 0.8.4)
       Build-Depends: semigroups == 0.18.*
       Build-Depends: transformers == 0.4.2.*
       Build-Depends: fail == 4.9.*
-- 
2.16.2.windows.1

