From c0f96635a62fde604f12bb21fe52caa0a0ad290c Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Fri, 6 Jan 2017 22:37:33 +0530
Subject: [PATCH] Patched

---
 distributive.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/distributive.cabal b/distributive.cabal
index ccf4150..28b0836 100644
--- a/distributive.cabal
+++ b/distributive.cabal
@@ -12,7 +12,7 @@ bug-reports:   http://github.com/ekmett/distributive/issues
 copyright:     Copyright (C) 2011-2016 Edward A. Kmett
 synopsis:      Distributive functors -- Dual to Traversable
 description:   Distributive functors -- Dual to Traversable
-build-type:    Custom
+build-type:    Simple
 extra-source-files:
   .travis.yml
   .vim.custom
-- 
2.7.4 (Apple Git-66)

