From 73750797b684cf1f341a93ac75ed96bf07885fa3 Mon Sep 17 00:00:00 2001
From: Jyothsna Srinivas <jyothsnasrinivas17@gmail.com>
Date: Sun, 29 Jul 2018 15:50:06 +0530
Subject: [PATCH] Patched

---
 Data/FileEmbed.hs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Data/FileEmbed.hs b/Data/FileEmbed.hs
index bbd83ee..e38eaa6 100644
--- a/Data/FileEmbed.hs
+++ b/Data/FileEmbed.hs
@@ -1,6 +1,6 @@
 {-# LANGUAGE TemplateHaskell #-}
 {-# LANGUAGE CPP #-}
-{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE OverloadedStrings, PatternSynonyms #-}
 -- | This module uses template Haskell. Following is a simplified explanation of usage for those unfamiliar with calling Template Haskell functions.
 --
 -- The function @embedFile@ in this modules embeds a file into the executable
@@ -60,7 +60,7 @@ import Language.Haskell.TH.Syntax
     )
 import System.Directory (doesDirectoryExist, doesFileExist,
                          getDirectoryContents, canonicalizePath)
-import Control.Exception (throw, ErrorCall(..))
+import Control.Exception (throw, ErrorCall(..), pattern ErrorCall)
 import Control.Monad (filterM)
 import qualified Data.ByteString as B
 import qualified Data.ByteString.Char8 as B8
-- 
2.15.2 (Apple Git-101.1)

