From 40c0e2a716ba842bfacc345aadb6ba598e1dcd1d Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Mon, 2 Oct 2017 15:56:30 -0400
Subject: [PATCH] Patched

---
 hashtables.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hashtables.cabal b/hashtables.cabal
index 42a4da7..ed7b650 100644
--- a/hashtables.cabal
+++ b/hashtables.cabal
@@ -141,7 +141,7 @@ Flag sse42
 
 Flag portable
   Description: if on, use only pure Haskell code and no GHC extensions.
-  Default: False
+  Default: True
 
 
 Library
-- 
2.7.4 (Apple Git-66)

