From 57d03352cd155a3dc9ff7380e6b9635b949b42d2 Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Tue, 16 May 2017 09:16:37 +0530
Subject: [PATCH] Patched

---
 http-api-data.cabal | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/http-api-data.cabal b/http-api-data.cabal
index e934369..b5c0d59 100644
--- a/http-api-data.cabal
+++ b/http-api-data.cabal
@@ -10,7 +10,7 @@ homepage:        http://github.com/fizruk/http-api-data
 category:        Web
 stability:       unstable
 cabal-version:   >= 1.10
-build-type:      Custom
+build-type:      Simple
 extra-source-files:
   include/overlapping-compat.h
   test/*.hs
@@ -22,11 +22,11 @@ tested-with:
   GHC==8.0.2,
   GHC==8.1.*
 
-custom-setup
-  setup-depends:
-    base >= 4.7 && <4.11,
-    Cabal >= 1.18 && <2.1,
-    cabal-doctest >=1.0.1 && <1.1
+-- custom-setup
+--   setup-depends:
+--     base >= 4.7 && <4.11,
+--     Cabal >= 1.18 && <2.1,
+--     cabal-doctest >=1.0.1 && <1.1
 
 flag use-text-show
   description: Use text-show library for efficient ToHttpApiData implementations.
-- 
2.7.4 (Apple Git-66)

