From b2e0cabe94433754f5822d00b30fe0ffb9b7a146 Mon Sep 17 00:00:00 2001
From: jneira <atreyu.bbb@gmail.com>
Date: Mon, 16 Apr 2018 21:34:48 +0200
Subject: [PATCH] Patched

---
 http-client.cabal | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/http-client.cabal b/http-client.cabal
index 6eae068..ae29c0d 100644
--- a/http-client.cabal
+++ b/http-client.cabal
@@ -70,8 +70,8 @@ library
   if impl(ghc < 7.10)
     buildable: False
 
-  if os(mingw32)
-    build-depends: Win32, safe
+  -- if os(mingw32)
+  --   build-depends: Win32, safe
 
   default-language:    Haskell2010
 
-- 
2.16.2.windows.1

