From f2351145c5e839984ded4d329a1c4035891f8100 Mon Sep 17 00:00:00 2001
From: monser <ouromoros@gmail.com>
Date: Tue, 5 Jun 2018 22:16:46 +0800
Subject: [PATCH] Patched

---
 Network/HPACK/Huffman/ByteString.hs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Network/HPACK/Huffman/ByteString.hs b/Network/HPACK/Huffman/ByteString.hs
index 96d501f..067650b 100644
--- a/Network/HPACK/Huffman/ByteString.hs
+++ b/Network/HPACK/Huffman/ByteString.hs
@@ -47,7 +47,7 @@ copy dst (PS fptr off len) = withForeignPtr fptr $ \ptr -> do
     let beg = ptr `plusPtr` off
     memcpy dst beg (fromIntegral len)
 
-foreign import ccall unsafe "string.h memcpy" c_memcpy
+foreign import java unsafe "@static eta.base.Utils.c_memcpy" c_memcpy
     :: Ptr Word8 -> Ptr Word8 -> CSize -> IO (Ptr Word8)
 
 memcpy :: Ptr Word8 -> Ptr Word8 -> Int -> IO ()
-- 
2.17.1

