From c552892ff8871313e4bc08aeb1996dabeb498cb9 Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Fri, 30 Dec 2016 02:10:51 +0530
Subject: [PATCH] Patched

---
 lens.cabal | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lens.cabal b/lens.cabal
index a2bf1da..375a525 100644
--- a/lens.cabal
+++ b/lens.cabal
@@ -11,7 +11,7 @@ stability:     provisional
 homepage:      http://github.com/ekmett/lens/
 bug-reports:   http://github.com/ekmett/lens/issues
 copyright:     Copyright (C) 2012-2016 Edward A. Kmett
-build-type:    Custom
+build-type:    Simple
 -- build-tools:   cpphs
 tested-with:   GHC == 7.4.2, GHC == 7.6.3, GHC == 7.8.4, GHC == 7.10.1, GHC == 7.10.2, GHC == 8.0.1
 synopsis:      Lenses, Folds and Traversals
@@ -299,7 +299,7 @@ library
   other-modules:
     Paths_lens

-  cpp-options: -traditional
+  -- cpp-options: -traditional

   if flag(safe)
     cpp-options: -DSAFE=1
--
2.7.4 (Apple Git-66)
