From 4553d456e6dfe1cac21c7c24cd633ebb89e6bb08 Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Wed, 16 Aug 2017 14:13:44 -0400
Subject: [PATCH] Patched

---
 linear.cabal | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/linear.cabal b/linear.cabal
index dadf5ae..0f289f6 100644
--- a/linear.cabal
+++ b/linear.cabal
@@ -13,7 +13,7 @@ bug-reports:   http://github.com/ekmett/linear/issues
 copyright:     Copyright (C) 2012-2015 Edward A. Kmett
 synopsis:      Linear Algebra
 description:   Types and combinators for linear algebra on free vector spaces
-build-type:    Custom
+build-type:    Simple
 tested-with:   GHC == 7.4.2, GHC == 7.6.3, GHC == 7.8.4, GHC == 7.10.3, GHC==8.0.2, GHC==8.2.1
 extra-source-files:
   .gitignore
@@ -30,8 +30,8 @@ flag template-haskell
     You can disable the use of the `template-haskell` package using `-f-template-haskell`.
     .
     Disabling this is an unsupported configuration, but it may be useful for accelerating builds in sandboxes for expert users.
-  default: True
-  manual: True
+  default: False
+  manual: False
 
 flag herbie
   description: Enable `herbie`.
@@ -42,11 +42,11 @@ source-repository head
   type: git
   location: https://github.com/ekmett/linear
 
-custom-setup
-  setup-depends:
-    base          >= 4 && < 5,
-    Cabal,
-    cabal-doctest >= 1 && < 1.1
+-- custom-setup
+--   setup-depends:
+--     base          >= 4 && < 5,
+--     Cabal,
+--     cabal-doctest >= 1 && < 1.1
 
 library
   build-depends:
-- 
2.7.4 (Apple Git-66)

