From 6bee2327a786f64f72af0fb732fb4790c683f666 Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Mon, 15 Jan 2018 18:21:21 +0530
Subject: [PATCH] Patched

---
 monad-logger.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/monad-logger.cabal b/monad-logger.cabal
index 7eadbde..fc1fe7a 100644
--- a/monad-logger.cabal
+++ b/monad-logger.cabal
@@ -19,7 +19,7 @@ source-repository head
 
 flag template_haskell {
       Description: Enable Template Haskell support
-      Default:     True
+      Default:     False
       Manual:      True
 }
 
-- 
2.7.4 (Apple Git-66)

