From 94c1a8794eb0a67652db478c0804fc6f6bc948c4 Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Sun, 25 Mar 2018 23:55:33 +0530
Subject: [PATCH] Patched

---
 Network/URI.hs | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Network/URI.hs b/Network/URI.hs
index db68f06..dcb36f9 100644
--- a/Network/URI.hs
+++ b/Network/URI.hs
@@ -110,7 +110,7 @@ module Network.URI
     , normalizePathSegments
 
     -- * Deprecated functions
-    , parseabsoluteURI
+    -- , parseabsoluteURI
     , escapeString
     , reserved, unreserved
     , scheme, authority, path, query, fragment
@@ -1254,9 +1254,9 @@ normalizePathSegments uristr = normstr juri
 --  Deprecated functions
 ------------------------------------------------------------
 
-{-# DEPRECATED parseabsoluteURI "use parseAbsoluteURI" #-}
-parseabsoluteURI :: String -> Maybe URI
-parseabsoluteURI = parseAbsoluteURI
+-- {-# DEPRECATED parseabsoluteURI "use parseAbsoluteURI" #-}
+-- parseabsoluteURI :: String -> Maybe URI
+-- parseabsoluteURI = parseAbsoluteURI
 
 {-# DEPRECATED escapeString "use escapeURIString, and note the flipped arguments" #-}
 escapeString :: String -> (Char->Bool) -> String
-- 
2.7.4 (Apple Git-66)

