From 8c6d9d2b88c4c3e30fe5023aaa7f1cc056362cd6 Mon Sep 17 00:00:00 2001
From: Jyothsna Srinivas <jyothsnasrinivas17@gmail.com>
Date: Wed, 11 Jul 2018 17:21:49 +0530
Subject: [PATCH] Patched

---
 src/Text/Parsec/Prim.hs | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/Text/Parsec/Prim.hs b/src/Text/Parsec/Prim.hs
index 5d5d017..371d805 100644
--- a/src/Text/Parsec/Prim.hs
+++ b/src/Text/Parsec/Prim.hs
@@ -112,6 +112,8 @@ import Control.Monad.Error.Class
 import Text.Parsec.Pos
 import Text.Parsec.Error
 
+import qualified Data.Function as F
+
 unknownError :: State s u -> ParseError
 unknownError state        = newErrorUnknown (statePos state)
 
@@ -156,7 +158,7 @@ newtype ParsecT s u m a
 -- | Low-level unpacking of the ParsecT type. To run your parser, please look to
 -- runPT, runP, runParserT, runParser and other such functions.
 runParsecT :: Monad m => ParsecT s u m a -> State s u -> m (Consumed (m (Reply s u a)))
-runParsecT p s = unParser p s cok cerr eok eerr
+runParsecT p s = F.trampoline $ unParser p s cok cerr eok eerr
     where cok a s' err = return . Consumed . return $ Ok a s' err
           cerr err = return . Consumed . return $ Error err
           eok a s' err = return . Empty . return $ Ok a s' err
-- 
2.15.2 (Apple Git-101.1)

