From 508439b140e0fef74f0b3bc53b4191a00a5cf5fb Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Thu, 6 Oct 2016 05:38:38 +0530
Subject: [PATCH] Patched

---
 System/Random.hs | 1 -
 1 file changed, 1 deletion(-)

diff --git a/System/Random.hs b/System/Random.hs
index ab77274..1eaf9fe 100644
--- a/System/Random.hs
+++ b/System/Random.hs
@@ -37,7 +37,6 @@
 --
 -----------------------------------------------------------------------------

-#include "MachDeps.h"

 module System.Random
 	(
--
2.7.4 (Apple Git-66)
