From 2faecafc91bff7f43426539698c9f4a2057b6a41 Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Thu, 10 Aug 2017 08:17:51 -0400
Subject: [PATCH] Patched

---
 scientific.cabal | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scientific.cabal b/scientific.cabal
index a915ec0..f74853b 100644
--- a/scientific.cabal
+++ b/scientific.cabal
@@ -56,7 +56,7 @@ flag bytestring-builder
 
 flag integer-simple
   description: Use the integer-simple package instead of integer-gmp
-  default:     False
+  default:     True
 
 library
   exposed-modules:     Data.ByteString.Builder.Scientific
@@ -83,7 +83,7 @@ library
       build-depends: bytestring         >= 0.10.4 && < 0.11
 
   if flag(integer-simple)
-      build-depends: integer-simple
+      build-depends: integer
   else
       build-depends: integer-gmp
 
-- 
2.7.4 (Apple Git-66)

