From 206015596d5d81374ca8155d5b2edfce694a068e Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Wed, 14 Dec 2016 00:41:49 +0530
Subject: [PATCH] Patched

---
 semigroupoids.cabal | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/semigroupoids.cabal b/semigroupoids.cabal
index ea27c1a..c8324c8 100644
--- a/semigroupoids.cabal
+++ b/semigroupoids.cabal
@@ -10,7 +10,7 @@ stability:     provisional
 homepage:      http://github.com/ekmett/semigroupoids
 bug-reports:   http://github.com/ekmett/semigroupoids/issues
 copyright:     Copyright (C) 2011-2015 Edward A. Kmett
-build-type:    Custom
+build-type:    Simple
 tested-with:   GHC == 7.0.4, GHC == 7.2.2, GHC == 7.4.2, GHC == 7.6.3, GHC == 7.8.4, GHC == 7.10.3, GHC == 8.0.1
 synopsis:      Semigroupoids: Category sans id
 extra-source-files:
@@ -177,4 +177,3 @@ test-suite doctests
       doctest   >= 0.9.1 && < 0.12,
       directory >= 1.0,
       filepath
-
--
2.7.4 (Apple Git-66)
