From 9abcfd668e808aa59fac21371f5157f70f766546 Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Sat, 13 May 2017 13:15:31 +0530
Subject: [PATCH] Patched

---
 servant.cabal | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/servant.cabal b/servant.cabal
index 824dbe5..5bb4c21 100644
--- a/servant.cabal
+++ b/servant.cabal
@@ -16,7 +16,7 @@ author:              Servant Contributors
 maintainer:          haskell-servant-maintainers@googlegroups.com
 copyright:           2014-2016 Zalora South East Asia Pte Ltd, Servant Contributors
 category:            Servant Web
-build-type:          Custom
+build-type:          Simple
 cabal-version:       >=1.10
 tested-with:         GHC >= 7.8
 extra-source-files:
@@ -26,9 +26,9 @@ source-repository head
   type: git
   location: http://github.com/haskell-servant/servant.git
 
-custom-setup
-  setup-depends:
-     Cabal >=1.14, base, filepath, directory
+-- custom-setup
+--   setup-depends:
+--      Cabal >=1.14, base, filepath, directory
 
 library
   exposed-modules:
-- 
2.7.4 (Apple Git-66)

