From 7e898dcb4e4d33538df18b7c0cff29d398fb823a Mon Sep 17 00:00:00 2001
From: Jyothsna Srinivas <jyothsnasrinivas17@gmail.com>
Date: Sun, 22 Jul 2018 15:38:09 +0530
Subject: [PATCH] Patched

---
 src/Data/StateVar.hs | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/Data/StateVar.hs b/src/Data/StateVar.hs
index bb4ba2f..0c87181 100644
--- a/src/Data/StateVar.hs
+++ b/src/Data/StateVar.hs
@@ -12,7 +12,7 @@
 -- Module      :  Data.StateVar
 -- Copyright   :  (c) Edward Kmett 2014-2015, Sven Panne 2009-2018
 -- License     :  BSD3
--- 
+--
 -- Maintainer  :  Sven Panne <svenpanne@gmail.com>
 -- Stability   :  stable
 -- Portability :  portable
@@ -78,7 +78,9 @@ module Data.StateVar
   ) where
 
 import Control.Concurrent.STM
+#if MIN_VERSION_base(4,11,0)
 import Control.Monad.IO.Class
+#endif
 import Data.IORef
 import Data.Typeable
 import Foreign.Ptr
-- 
2.15.2 (Apple Git-101.1)

