From 45fdfc7c04eb1041d76fc29487591954d502b326 Mon Sep 17 00:00:00 2001
From: Jyothsna Srinivas <jyothsnasrinivas17@gmail.com>
Date: Tue, 17 Jul 2018 15:34:25 +0530
Subject: [PATCH] Patched

---
 transformers.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/transformers.cabal b/transformers.cabal
index 8583b12..88960b2 100644
--- a/transformers.cabal
+++ b/transformers.cabal
@@ -46,7 +46,7 @@ library
     -- NB: using impl(ghc>=7.9) instead of fragile Cabal flags
     hs-source-dirs: legacy/pre709
     exposed-modules: Data.Functor.Identity
-  if !impl(ghc>=7.11)
+  if !impl(ghc>=7.11) && impl(eta<0.8.4)
     -- modules moved into base-4.9.0 (GHC 8.0)
     -- see https://ghc.haskell.org/trac/ghc/ticket/10773
     -- see https://ghc.haskell.org/trac/ghc/ticket/11135
-- 
2.15.2 (Apple Git-101.1)

