From 3587ebce9648754d1cc78cdc541235530e9635d4 Mon Sep 17 00:00:00 2001
From: Jyothsna Srinivas <jyothsnasrinivas17@gmail.com>
Date: Sun, 22 Jul 2018 14:07:18 +0530
Subject: [PATCH] Patched

---
 Control/Monad/Trans/Error.hs | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Control/Monad/Trans/Error.hs b/Control/Monad/Trans/Error.hs
index 7d01910..bbaf55e 100644
--- a/Control/Monad/Trans/Error.hs
+++ b/Control/Monad/Trans/Error.hs
@@ -66,6 +66,8 @@ import Data.Monoid (mempty)
 import Data.Traversable (Traversable(traverse))
 import System.IO.Error
 
+#if !MIN_VERSION_base(4,10,0)
+
 instance MonadPlus IO where
     mzero       = ioError (userError "mzero")
     m `mplus` n = m `catchIOError` \ _ -> n
@@ -74,6 +76,8 @@ instance Alternative IO where
     empty = mzero
     (<|>) = mplus
 
+#endif    
+
 #if !(MIN_VERSION_base(4,4,0))
 -- exported by System.IO.Error from base-4.4
 catchIOError :: IO a -> (IOError -> IO a) -> IO a
-- 
2.15.2 (Apple Git-101.1)

