From 1c1cc0533447c4874a77848bd8bc3cbf8b691aff Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Sun, 25 Mar 2018 20:14:05 +0530
Subject: [PATCH] Patched

---
 Data/Vector.hs | 2 +-
 vector.cabal   | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/Data/Vector.hs b/Data/Vector.hs
index 21b6196..51df664 100644
--- a/Data/Vector.hs
+++ b/Data/Vector.hs
@@ -166,7 +166,7 @@ module Data.Vector (
 
 import qualified Data.Vector.Generic as G
 import           Data.Vector.Mutable  ( MVector(..) )
-import           Data.Primitive.Array
+import           Data.Primitive.Array hiding (fromList, fromListN)
 import qualified Data.Vector.Fusion.Bundle as Bundle
 
 import Control.DeepSeq ( NFData, rnf )
diff --git a/vector.cabal b/vector.cabal
index 38c957d..3a4f68e 100644
--- a/vector.cabal
+++ b/vector.cabal
@@ -152,6 +152,8 @@ Library
     Build-Depends: semigroups >= 0.18 && < 0.19
 
   Ghc-Options: -O2 -Wall
+  if impl(eta < 0.7.1.3)
+    Ghc-Options: -optP--text
 
   if !flag(Wall)
     Ghc-Options: -fno-warn-orphans
-- 
2.7.4 (Apple Git-66)

