From aed44b0c3f59f1cc8000292d53beec1817bb0a67 Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Sat, 31 Mar 2018 18:10:00 +0530
Subject: [PATCH] Patched

---
 Data/Vector.hs | 2 +-
 vector.cabal   | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/Data/Vector.hs b/Data/Vector.hs
index 21b6196..51df664 100644
--- a/Data/Vector.hs
+++ b/Data/Vector.hs
@@ -166,7 +166,7 @@ module Data.Vector (
 
 import qualified Data.Vector.Generic as G
 import           Data.Vector.Mutable  ( MVector(..) )
-import           Data.Primitive.Array
+import           Data.Primitive.Array hiding (fromList, fromListN)
 import qualified Data.Vector.Fusion.Bundle as Bundle
 
 import Control.DeepSeq ( NFData, rnf )
diff --git a/vector.cabal b/vector.cabal
index 8f0b57f..329f1b2 100644
--- a/vector.cabal
+++ b/vector.cabal
@@ -158,6 +158,9 @@ Library
 
   Ghc-Options: -O2 -Wall
 
+  if impl(eta < 0.7.1.3)
+    Ghc-Options: -optP--text
+
   if !flag(Wall)
     Ghc-Options: -fno-warn-orphans
 
-- 
2.7.4 (Apple Git-66)

