From a789da6495a5c68e7875669f4a17c2f7de9a7628 Mon Sep 17 00:00:00 2001
From: Rahul Muttineni <rahulmutt@gmail.com>
Date: Mon, 13 Nov 2017 06:05:50 +0530
Subject: [PATCH] Patched

---
 Data/Vector.hs | 2 +-
 vector.cabal   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Data/Vector.hs b/Data/Vector.hs
index db6dfb7..d3292d0 100644
--- a/Data/Vector.hs
+++ b/Data/Vector.hs
@@ -161,7 +161,7 @@ module Data.Vector (
 
 import qualified Data.Vector.Generic as G
 import           Data.Vector.Mutable  ( MVector(..) )
-import           Data.Primitive.Array
+import           Data.Primitive.Array hiding (fromList, fromListN)
 import qualified Data.Vector.Fusion.Bundle as Bundle
 
 import Control.DeepSeq ( NFData, rnf )
diff --git a/vector.cabal b/vector.cabal
index ffc5099..47dbdd7 100644
--- a/vector.cabal
+++ b/vector.cabal
@@ -142,7 +142,7 @@ Library
                , ghc-prim >= 0.2 && < 0.5
                , deepseq >= 1.1 && < 1.5
 
-  Ghc-Options: -O2 -Wall -fno-warn-orphans
+  Ghc-Options: -O2 -Wall -fno-warn-orphans -optP--text
 
   if flag(BoundsChecks)
     cpp-options: -DVECTOR_BOUNDS_CHECKS
-- 
2.7.4 (Apple Git-66)

